src/axolotl/core/trainer_builder.py

```diff
@@ -997,6 +999,14 @@ def tokenize_row(
                 res[key] = res[key][1:]
         return res
 
+    def training_step(
+        self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor, Any]]
+    ) -> torch.Tensor:
+        loss: torch.Tensor = super().training_step(model, inputs)
+        torch.cuda.empty_cache()
+        gc.collect()
```

Comment:
Shouldn't gc.collect be first? 
In my experience if there is a true leak (hanging references) empty cache cannot not get rid of leaks

---

If it helps, I can try and debug where the leak is given the training config

Reply:
Thanks. I reordered the calls and added some context from reproduction